libs/datasets/timeseries.py

```diff
@@ -1571,3 +1602,42 @@ def make_source_url_tags(ds_in: MultiRegionDataset) -> MultiRegionDataset:
     )
     source_url[TagField.TYPE] = TagType.SOURCE_URL
     return ds_in.append_tag_df(source_url)
+
+
+T_SERIES_OR_DATAFRAME = TypeVar("T_SERIES_OR_DATAFRAME", pd.Series, pd.DataFrame)
```

Comment:
nit: I thought `T_SERIES` was a weird abbreviation of TIMESERIES at first glance.  I don't think we use `T_` as a type prefix anywhere else in the codebase.  And in general I think the convention for type variables is to use CamelCase (e.g. `AnyStr`, `Dict`, `Union`, etc.).  And we actually have an identical `SeriesOrDataframe` type variable already over here: https://github.com/covid-projections/covid-data-model/blob/769738a9da44ad9fd98518bf869042bd222cad73/libs/parallel_utils.py#L28 (though I'm the one that added it so who knows if I did it optimally 😁 ).

Reply:
I'm 80% sure this is not an intended use of pandas but `from pandas.core.generic import FrameOrSeries` does what we want and is heavily used within pandas. _typing.py has a nice comment explaining TypeVar vs Union. See also https://stackoverflow.com/a/58775376